proto/frontend/PI/frontends/proto/config_mgr.h

```diff
@@ -0,0 +1,62 @@
+/* Copyright 2013-present Barefoot Networks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Antonin Bas (antonin@barefootnetworks.com)
+ *
+ */
+
+#ifndef PI_FRONTENDS_PROTO_CONFIG_MGR_H_
+#define PI_FRONTENDS_PROTO_CONFIG_MGR_H_
+
+#include <memory>
+
+#include "gnmi/gnmi.pb.h"
+#include "google/rpc/status.pb.h"
+
+namespace pi {
+
+namespace fe {
+
+namespace proto {
+
+// forward declaration for PIMPL class
+class GnmiConfigMgrImp;
+
+class GnmiConfigMgr {
```

Comment:
nit: The gNMI interface will be used for more than just Config. The other use-cases are (i) streaming telemetry (ii) streaming notifications to controller (iii) run-time control from controller.
Given that, perhaps call this GnmiMgr to not suggest that this is config-specific (unless you plan to create separate classes and plumbing for other use-cases).

gNMIServiceImpl gnmi_service